app/controllers/session_controller.rb

```diff
@@ -663,32 +663,27 @@ def current
   end
 
   def destroy
-    redirect_url = params[:return_url].presence || SiteSetting.logout_redirect.presence
-
-    sso = SiteSetting.enable_discourse_connect
-    only_one_authenticator =
-      !SiteSetting.enable_local_logins && Discourse.enabled_authenticators.length == 1
-    if SiteSetting.login_required && (sso || only_one_authenticator)
-      # In this situation visiting most URLs will start the auth process again
-      # Go to the `/login` page to avoid an immediate redirect
-      redirect_url ||= path("/login")
-    end
-
+    redirect_url = params[:return_url].presence
+    redirect_url ||= SiteSetting.logout_redirect.presence
     redirect_url ||= path("/")
```

Comment:
In my testing, visiting `/` on a login-required site with only one IDP also triggers the auto redirect behaviour 🤔